src/SFML/Graphics/ConvexShape.cpp

```diff
@@ -66,4 +66,44 @@ Vector2f ConvexShape::getPoint(std::size_t index) const
     return m_points[index];
 }
 
+////////////////////////////////////////////////////////////
+Vector2f ConvexShape::getGeometricCenter() const
+{
+    if (m_points.size() > 2)
+    {
+        const auto& [offx, offy] = m_points.front();
+        float x = 0.f, y = 0.f;
+        float twicearea = 0.f;
+
+        float f;
+
+        for (auto it = m_points.begin(), prev = std::prev(m_points.end()); it != m_points.end(); prev = it++)
+        {
+            const auto& [p1x, p1y] = *it;
+            const auto& [p2x, p2y] = *prev;
+
+            f = (p1x - offx) * (p2y - offy) - (p2x - offx) * (p1y - offy);
+            twicearea += f;
+            x += (p1x + p2x - 2.f * offx) * f;
+            y += (p1y + p2y - 2.f * offy) * f;
+        }
```

Comment:
```suggestion
        for (std::size_t i = 0; i < m_points.size() - 1; ++i)
        {
            const auto& [p1x, p1y] = m_points[i];
            const auto& [p2x, p2y] = m_points[i + 1];

            f = (p1x - offx) * (p2y - offy) - (p2x - offx) * (p1y - offy);
            twicearea += f;
            x += (p1x + p2x - 2.f * offx) * f;
            y += (p1y + p2y - 2.f * offy) * f;
        }
```

Honestly I find an index-based version easier and simpler to read. Also, we should use vector operations in the body (i.e. instead of manually acting on `x` and `y`, use a `Vector2f` for your result).

Reply:
@vittorioromeo This will however not include the line between the last point and the first point - but I can rewrite it using idices instead.